### PR TITLE
Javascript Modules

### PR DESCRIPTION
**Pull request by DanielMwendwa**
Why use modules? Maintainability,  Namespacing (modules allow us to avoid namespace pollution by creating a private space for our variables), Reusability. more info on [modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules)

### CODE REVIEW

**Review by IsaackMwenda — CHANGES_REQUESTED** [review]
Hi Daniel please resolve conflicts

**Comment by DanielMwendwa** [comment]
conflicts resolved

File: public/app.js [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+import { DataController } from "./data.js";
+
 // GLOBAL APP CONTROLLER
 class Controller {
     static setupEventListeners() {
         Controller.DOMstrings = UIController.getDOMstrings();
         document
             .querySelector(Controller.DOMstrings.logoutBtn)
-            .addEventListener("click", AuthController.logout);
+            .addEventListener("click", () => {import("./auth.js").then(module => {
```
Comment (IsaackMwenda) [inline public/app.js]: Why are dynamically importing this? It is used twice so cleaner if imported statically.

File: public/graph.js [hunk 2]
```diff
@@ -0,0 +1,1294 @@
+// GRAPH CONTROLLER
+export class GraphController {
```
Comment (IsaackMwenda) [inline public/graph.js]: please delete before merging.

**Review by IsaackMwenda — APPROVED** [review]

File: public/app.js (hunk 1 above)
Comment (DanielMwendwa) [reply public/app.js]: Updated thanks.

File: public/graph.js (hunk 2 above)
Comment (DanielMwendwa) [reply public/graph.js]: Deleted  graph.js file

**Pull request merged**